ui/app/pages/confirm-token-transaction-base/confirm-token-transaction-base.component.js

```diff
@@ -11,30 +13,36 @@ import {
 import { getWeiHexFromDecimalValue } from '../../helpers/utils/conversions.util'
 import { ETH, PRIMARY } from '../../helpers/constants/common'
 
-export default class ConfirmTokenTransactionBase extends Component {
-  static contextTypes = {
-    t: PropTypes.func,
-  }
+const ConfirmTokenTransactionBase = ({
+  toAddress,
+  tokenAddress,
+  tokenAmount = '0',
+  tokenSymbol,
+  fiatTransactionTotal,
+  ethTransactionTotal,
+  contractExchangeRate,
+  conversionRate,
+  currentCurrency,
+}) => {
 
-  static propTypes = {
-    tokenAddress: PropTypes.string,
-    toAddress: PropTypes.string,
-    tokenAmount: PropTypes.string,
-    tokenSymbol: PropTypes.string,
-    fiatTransactionTotal: PropTypes.string,
-    ethTransactionTotal: PropTypes.string,
-    contractExchangeRate: PropTypes.number,
-    conversionRate: PropTypes.number,
-    currentCurrency: PropTypes.string,
-  }
+  const t = useContext(I18nContext)
+  const hexWeiValue = useMemo(() => {
+    if (tokenAmount === '0' || !contractExchangeRate) {
+      return '0'
+    }
 
-  static defaultProps = {
-    tokenAmount: '0',
-  }
+    const decimalEthValue = (
+      (new BigNumber(tokenAmount)).mul(new BigNumber(contractExchangeRate))
```

Comment:
Nit: Perhaps we should use `.times` in place of `.mul`? Both are valid in the version of `bignumber.js` we're using (`v4.1.0`), but `.mul` was removed in `v6.0.0` whereas `.times` is still supported in the latest version.
```suggestion
      (new BigNumber(tokenAmount)).times(new BigNumber(contractExchangeRate))
```